### DISCUSSION TITLE
toolkit v0.1: Upgrade to the v1beta1 APIs

### DISCUSSION THREAD

**Discussion by stefanprodan** [fluxcd/flux2#288] [Show and tell]
GOTK v0.1 the first MINOR prerelease, it promotes the `toolkit.fluxcd.io` APIs to v1beta1 and removes support for v1alpha1.

Going forward, changes to the API will be accompanied by a conversion mechanism. With this release the API becomes more stable, but while in beta phase there are no guarantees about backwards compatibility between beta releases.

### Upgrade procedure from 0.0.x to 0.1.x

- install the latest v0.1 gotk CLI `brew upgrade gotk`
- uninstall v0.0.28 with `gotk uninstall --crds --namespace=gitops-system` (this will wipe clean your cluster)
- delete the gitops-system dir from Git
- install v0.1 `gotk install --namespace=gotk-system` or using `gotk bootstrap` 
- update the toolkit custom resources in from `toolkit.fluxcd.io/v1alpha1` to `toolkit.fluxcd.io/v1beta1`

### Manual removal of 0.0.x

Assuming you've installed the toolkit in `gitops-system`, you could uninstall it without removing the deployed workloads:

* delete the controllers
```sh
kubectl -n gitops-system delete deployments --all
```
* remove the finalizers for each kind (kustomization, gitrepositry, helmrelease, helmrepository):
```sh
kubectl -n gitops-system patch kustomization <name> -p '{"metadata":{"finalizers":null}}
```
* delete the namespace
```sh
kubectl delete ns gitops-system
```
* delete the crds
```sh
kubectl delete $(kubectl get crds -oname | grep "toolkit.fluxcd.io")
```

**Comment by allymparker** [comment]
Don't forget to change your manifest API versions:
`apiVersion: helm.toolkit.fluxcd.io/v2alpha1` -> `apiVersion: helm.toolkit.fluxcd.io/v2beta1`
`apiVersion: kustomize.toolkit.fluxcd.io/v1alpha1` -> `apiVersion: kustomize.toolkit.fluxcd.io/v1beta1`

Delete your `gitops-system` folders from git

Namespaces from `gitops-system` -> `gotk-system`
